keepassxc-browser/content/custom-fields-banner.js

```diff
@@ -626,138 +618,120 @@ kpxcCustomLoginFieldsBanner.getNonSelectedElements = function() {
-    const frameUrl = new URL(window.self.document.location.origin);
-    const urlParts = originUrl.host.split('.');
-    const dotCount = urlParts.length - 1;
-
-    // Simple host like google.com, check directly
-    if (dotCount < 1) {
-        return false;
-    } else if (dotCount === 1) {
-        return frameUrl.host.includes(originUrl.host);
-    }
-
-    // Get the top-level-domain using counts of '.' but backwards, max 3.
-    // A basic host is like idmsa.apple.com, a more complex one like www.bbva.com.ar.
-    const index = Math.min(dotCount, 3);
-    const subDomain = `${urlParts[dotCount - index]}.`;
-    const topLevelDomain = originUrl.host.substring(originUrl.host.indexOf(subDomain) + subDomain.length);
-
-    return frameUrl.host.includes(topLevelDomain);
-};
-
 // Handles messages sent from iframes to the top window
-const handleTopWindowMessage = function(e) {
-    if (!topLevelDomainMatches(e.origin)) {
+kpxcCustomLoginFieldsBanner.handleTopWindowMessage = function(args) {
+    if (!kpxcCustomLoginFieldsBanner.created) {
         return;
     }
 
-    if (e.data.message === 'username_selected') {
-        kpxcCustomLoginFieldsBanner.selection.username = e.data.selection;
+    const message = args?.[2];
+    const selection = args?.[3];
+
+    if (message === 'username_selected') {
+        kpxcCustomLoginFieldsBanner.selection.username = selection;
         kpxcCustomLoginFieldsBanner.setSelectedField();
-    } else if (e.data.message === 'password_selected') {
-        kpxcCustomLoginFieldsBanner.selection.password = e.data.selection;
+    } else if (message === 'password_selected') {
+        kpxcCustomLoginFieldsBanner.selection.password = selection;
         kpxcCustomLoginFieldsBanner.setSelectedField();
-    } else if (e.data.message === 'totp_selected') {
-        kpxcCustomLoginFieldsBanner.selection.totp = e.data.selection;
+    } else if (message === 'totp_selected') {
+        kpxcCustomLoginFieldsBanner.selection.totp = selection;
         kpxcCustomLoginFieldsBanner.setSelectedField();
-    } else if (e.data.message === 'string_field_selected') {
-        kpxcCustomLoginFieldsBanner.selection.stringFields = e.data.selection;
+    } else if (message === 'string_field_selected') {
+        kpxcCustomLoginFieldsBanner.selection.stringFields = selection;
         kpxcCustomLoginFieldsBanner.setSelectedField();
-    } else if (e.data.message === 'enable_clear_data_button') {
+    } else if (message === 'enable_clear_data_button') {
         kpxcCustomLoginFieldsBanner.buttons.clearData.style.display = 'inline-block';
     }
 };
 
 // Handle Banner button clicks from the top window
-const handleParentWindowMessage = function(e) {
-    if (!topLevelDomainMatches(e.origin)) {
+kpxcCustomLoginFieldsBanner.handleParentWindowMessage = function(args) {
+    if (!kpxcCustomLoginFieldsBanner.created) {
         return;
     }
 
-    if (e.data === 'username_button_clicked') {
+    const e = {};
+    e.isTrusted = args?.[1];
```

Comment:
This is the only value that's needed, but some wrapping is needed so it's compatible with an actual Event. The whole event cannot be cloned and sent to background script.